Update platform toolset for C++ projects

To build PTVS on latest VS Preview 2 builds, I had to install an additional component for the Visual Studio 2017 (v141) toolset, since v142 is the one that gets installed now.

We should change our projects to target v142. I do not know of any reason to stay on v141, but there might be?

![image](https://user-images.githubusercontent.com/1696845/49675451-a34b6d80-fa2a-11e8-9abe-ced7a3ac4461.png)

Edit: Same for the Windows SDK:
```
1>------ Build started: Project: DebuggerHelper, Configuration: Debug x64 ------
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Microsoft\VC\v150\Microsoft.Cpp.WindowsSDK.targets(46,5): error MSB8036: The Windows SDK version 10.0.17134.0 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution".
1>Done building project "DebuggerHelper.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

10.0.17763.0 is the latest/default install.

--- a/Build/Common.Build.Cpp.settings
+++ b/Build/Common.Build.Cpp.settings
@@ -102,7 +102,8 @@
   <PropertyGroup>
     <_WinSdkRegistryVersion>$(Registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SDKs\Windows\v10.0@ProductVersion)</_WinSdkRegistryVersion>
     <_WinSdkRegistryVersion Condition="$(_WinSdkRegistryVersion) == ''">$(Registry:HKEY_LOCAL_MACHINE\WOW6432Node\SOFTWARE\Microsoft\Microsoft SDKs\Windows\v10.0@ProductVersion)</_WinSdkRegistryVersion>
-    <DefaultWindowsSDKVersion>10.0.17134.0</DefaultWindowsSDKVersion>
+    <DefaultWindowsSDKVersion>10.0.17763.0</DefaultWindowsSDKVersion>
+    <DefaultWindowsSDKVersion Condition="$(_WinSdkRegistryVersion) == '10.0.17134'">10.0.17134.0</DefaultWindowsSDKVersion>
     <DefaultWindowsSDKVersion Condition="$(_WinSdkRegistryVersion) == '10.0.16299'">10.0.16299.0</DefaultWindowsSDKVersion>
     <DefaultWindowsSDKVersion Condition="$(_WinSdkRegistryVersion) == '10.0.15063'">10.0.15063.0</DefaultWindowsSDKVersion>
     <DefaultWindowsSDKVersion Condition="$(_WinSdkRegistryVersion) == '10.0.14393'">10.0.14393.0</DefaultWindowsSDKVersion>

--- a/Python/Product/DebuggerHelper/DebuggerHelper.vcxproj
+++ b/Python/Product/DebuggerHelper/DebuggerHelper.vcxproj
@@ -50,7 +50,7 @@
     <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Label="Configuration" Condition=" '$(VisualStudioVersion)'=='16.0'  Or '$(TargetVisualStudioVersion)'=='VS160' ">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">

--- a/Python/Product/DebuggerHelper/DebuggerHelperX86.vcxproj
+++ b/Python/Product/DebuggerHelper/DebuggerHelperX86.vcxproj
@@ -50,7 +50,7 @@
     <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Label="Configuration" Condition=" '$(VisualStudioVersion)'=='16.0'  Or '$(TargetVisualStudioVersion)'=='VS160' ">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">

--- a/Python/Product/PyDebugAttach/PyDebugAttach.vcxproj
+++ b/Python/Product/PyDebugAttach/PyDebugAttach.vcxproj
@@ -49,7 +49,7 @@
     <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Label="Configuration" Condition=" '$(VisualStudioVersion)'=='16.0'  Or '$(TargetVisualStudioVersion)'=='VS160' ">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">

--- a/Python/Product/PyDebugAttach/PyDebugAttachX86.vcxproj
+++ b/Python/Product/PyDebugAttach/PyDebugAttachX86.vcxproj
@@ -49,7 +49,7 @@
     <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Label="Configuration" Condition=" '$(VisualStudioVersion)'=='16.0'  Or '$(TargetVisualStudioVersion)'=='VS160' ">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">

--- a/Python/Product/VsPyProf/VsPyProf.vcxproj
+++ b/Python/Product/VsPyProf/VsPyProf.vcxproj
@@ -18,7 +18,6 @@
     <ProjectGuid>{6EC75EDA-17AC-453E-BA34-CB6796C4319B}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
     <RootNamespace>VsPyProf</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
   </PropertyGroup>
   <Choose>
     <When Condition=" '$(VisualStudioVersion)'=='15.0'  Or '$(TargetVisualStudioVersion)'=='VS150' ">

--- a/Python/Product/VsPyProf/VsPyProf.vcxproj
+++ b/Python/Product/VsPyProf/VsPyProf.vcxproj
@@ -18,6 +18,7 @@
     <ProjectGuid>{6EC75EDA-17AC-453E-BA34-CB6796C4319B}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
     <RootNamespace>VsPyProf</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
   </PropertyGroup>
   <Choose>
     <When Condition=" '$(VisualStudioVersion)'=='15.0'  Or '$(TargetVisualStudioVersion)'=='VS150' ">
@@ -49,7 +50,7 @@
     <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Label="Configuration" Condition=" '$(VisualStudioVersion)'=='16.0'  Or '$(TargetVisualStudioVersion)'=='VS160' ">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">

--- a/Python/Product/VsPyProf/VsPyProfX86.vcxproj
+++ b/Python/Product/VsPyProf/VsPyProfX86.vcxproj
@@ -49,7 +49,7 @@
     <PlatformToolset>v141</PlatformToolset>
   </PropertyGroup>
   <PropertyGroup Label="Configuration" Condition=" '$(VisualStudioVersion)'=='16.0'  Or '$(TargetVisualStudioVersion)'=='VS160' ">
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
   </PropertyGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.props" />
   <ImportGroup Label="ExtensionSettings">

